NUnitTests/Lessons/23/LambdaHomework.cs

```diff
@@ -52,73 +52,87 @@ public void Test_FilterEvenNumbers_Function()
 
 
         // TODO: Uncomment and implement lambda functions instead of regular functions
-        /*
         [Test]
         public void Test_Add_Function_To_Lambda()
         {
-            var lambda = 
+            Func<int, int, int> lambda = (x, y) => x + y;
             var result = lambda(3, 4);
+
             Assert.That(result, Is.EqualTo(7));
         }
 
         [Test]
         public void Test_Multiply_Function_To_Lambda()
         {
-            var lambda = 
+            
+            Func<int, int, int> lambda = (x, y) => x * y;
+
             var result = lambda(3, 4);
             Assert.That(result, Is.EqualTo(12));
         }
 
         [Test]
         public void Test_FilterEvenNumbers_Function_To_Lambda()
         {
-            var lambda =
+            
+            var lambda = (List<int> numbers) => numbers.Where(n => n % 2 == 0).ToList();
+
             var input = new List<int> { 1, 2, 3, 4 };
             var expected = new List<int> { 2, 4 };
             var result = lambda(input);
+
             Assert.That(result, Is.EqualTo(expected).AsCollection);
         }
-        */
+        
         [Test]
         public void Test_Where_Lambda()
         {
             var myList = new List<string> { "one", "two", "three", "four" };
             var filterredList = myList.Where(e => e.Contains('t'));
             Assert.That(filterredList.Count, Is.EqualTo(2));
         }
+
         // TODO: Uncomment and implement without lambda functions
-        /*
         [Test]
         public void Test_Where_NoLambda()
         {
             var myList = new List<string> { "one", "two", "three", "four" };
             var filterredList = new List<string>();
 
-            ...
-
+            foreach (var item in myList)
+            {
+                if (item.Length == 3)
+                {
+                    filterredList.Add(item);
+                }
+            }
             Assert.That(filterredList.Count, Is.EqualTo(2));
         }
-        */
+        
         [Test]
         public void Test_All_Lambda()
         {
             var myList = new List<string> { "one", "two", "three", "four" };
             var notEmpty = myList.All(e => e.Length > 0);
             Assert.That(notEmpty, Is.True);
         }
+
         // TODO: Uncomment and implement without lambda functions
-        /*
         [Test]
         public void Test_All_NoLambda()
         {
             var myList = new List<string> { "one", "two", "three", "four" };
-            bool result
-
-            ...
+            bool result = true;
 
+             foreach (var item in myList)
+            {
+                if (item.Length <= 2)
```

Comment:
the same here. I meant condition "Length > 0"

Reply:
@vldemyan i'll update the tests to be aligned with logic.

Reply:
fixed